Dage 79: (feat) Add Autosave to Datastore

Intrinsical-AI opened the pull request:
[TASK: DAGE-79](https://sease.atlassian.net/browse/DAGE-79)

Adds an optional auto-save feature to DataStore that periodically persists changes after N updates, enhancing data safety.

## Summary
- New config parameter: datastore_autosave_every_n_updates (default: disabled)
- Auto-save logic implementation
- Unit test expansion
-  Updated documentation

> Autosave off by default

Intrinsical-AI committed:
feat(text-clean-standarized): Expanded text cleaning function and unified in engines and datastore

Unified text cleaning funtion in commons/utils.py, used across engines and datastore (query deduplication). Added tests and updated existings.

Intrinsical-AI committed:
feat(text-clean-standarized): Expanded text cleaning function and unified in engines and datastore

Used new implementation for query deduplication in the datastore

Intrinsical-AI committed:
feat(text-clean-standarized): Expanded text cleaning function and unified in engines and datastore

Added and updated existing unit tests. Dedplication, text cleaning / utils included.

Intrinsical-AI committed:
feat(text-clean-standarized): Expanded text cleaning function and unified in engines and datastore

Import from commons. Standar usage of text cleaning across engines with new implementation

Intrinsical-AI committed:
feat(text-clean-standarized): Expanded text cleaning function and unified in engines and datastore

Removed old clean_text implementation, unfiied in commons.utils; added import for retro-compat in case old code still importing clean_text from dataset-generator module

Intrinsical-AI committed:
feat(text-clean-standarized): Expanded text cleaning function and unified in engines and datastore

Added related explaination on Solr default schema to README

Intrinsical-AI committed:
feat(text-clean-standarized): Expanded text cleaning function and unified in engines and datastore

Fixing CI checks (lint, tests)

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Added config field datastore_autosave_every_n_updates to control auto-saving every N changes. Additional logic on the datastore. Updated and expanded tests; and also updated config files accordingly.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Added config field datastore_autosave_every_n_updates to control auto-saving every N changes. Additional logic on the datastore. Updated and expanded tests; and also updated config files accordingly.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Added config field datastore_autosave_every_n_updates to control auto-saving every N changes. Additional logic on the datastore. Updated and expanded tests; and also updated config files accordingly.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Added config field datastore_autosave_every_n_updates to control auto-saving every N changes. Additional logic on the datastore. Updated and expanded tests; and also updated config files accordingly.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Added config field datastore_autosave_every_n_updates to control auto-saving every N changes. Additional logic on the datastore. Updated and expanded tests; and also updated config files accordingly.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Added config field datastore_autosave_every_n_updates to control auto-saving every N changes. Additional logic on the datastore. Updated and expanded tests; and also updated config files accordingly.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Added config field datastore_autosave_every_n_updates to control auto-saving every N changes. Additional logic on the datastore. Updated and expanded tests; and also updated config files accordingly.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Minor: removed import to pass linters.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Minor: removed import to pass linters.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Minor: removed import to pass linters.

Intrinsical-AI committed:
Merge branch 'DAGE-79' of https://github.com/SeaseLtd/rated-ranking-evaluator into DAGE-79

Intrinsical-AI committed:
Quick update addressing Nicolo's proposal

nicolo-rinaldi reviewed (commented):
It's better to update `test_config.py` file since you added a param to the config file
Also better to update the README file with the new param

nseidan commented on rre-tools/commons/src/commons/data_store.py:
```diff
@@ -144,6 +152,23 @@ def create_rating_score(
+                self.save()
+                log.debug(f"[autosave] ok path={self.path} updates={self._updates_since_last_save}")
+            except Exception as e:
+                log.warning(f"[autosave] fail err={e}")
```
In case of any exception, we don't save but in the finally block `self._updates_since_last_save = 0` always runs.

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Fix: Naz identified a potential bug on the autosave counter reset, now should be fixed: if we encounter any error when auto-saving, we skip without breaking execution

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

Improved: as Nicolo suggested, I expanded the config test cases to the new param

Intrinsical-AI committed:
Merge branch 'DAGE-79' of https://github.com/SeaseLtd/rated-ranking-evaluator into DAGE-79

Intrinsical-AI committed:
feat(datastore): implement optional auto-saving mechanism

fixed small bug arised from conflict solving
